lib/lotus/commands/new.rb

```diff
@@ -10,11 +11,11 @@ class New
       attr_reader :app_name, :source, :target, :cli, :options
 
       def initialize(app_name, environment, cli)
-        @app_name = app_name
+        @app_name = ApplicationName.new(app_name)
         @options  = environment.to_options
         @arch     = @options.fetch(:architecture)
 
-        @target   = Pathname.pwd.join(@app_name)
+        @target   = Pathname.pwd.join(@app_name.to_s)
```

Comment:
:+1: -- even if it's just right now aliased to `#to_s`, that's a good idea in terms of API surface.